Sources/TokamakCore/MountedViews/MountedHostView.swift

```diff
@@ -56,55 +62,89 @@ public final class MountedHostView<R: Renderer>: MountedElement<R> {
 
     guard !view.children.isEmpty else { return }
 
+    let isGroupView = view.type is GroupView.Type
+    // Don't allow children to transition their mounting since they aren't individually
+    // appearing (unless its a `GroupView`, which is flattened).
     mountedChildren = view.children.map {
-      $0.makeMountedView(reconciler.renderer, target, environmentValues, self)
+      $0.makeMountedView(
+        reconciler.renderer,
+        target,
+        environmentValues,
+        isGroupView ? self.viewTraits : .init(),
+        self
+      )
     }
 
     /* Remember that `GroupView`s are always "flattened", their `target` instances are targets of
      their parent elements. We need the insertion "cursor" `sibling` to be preserved when children
      are mounted in that case. Thus pass the `sibling` target to the children if `view` is a
      `GroupView`.
      */
-    let isGroupView = view.type is GroupView.Type
     mountedChildren.forEach {
-      $0.mount(before: isGroupView ? sibling : nil, on: self, with: reconciler)
+      $0.mount(before: isGroupView ? sibling : nil, on: self, in: reconciler, with: transaction)
     }
+
+    super.mount(before: sibling, on: parent, in: reconciler, with: transaction)
   }
 
-  override func unmount(with reconciler: StackReconciler<R>) {
+  private var parentUnmountTask = UnmountTask<R>()
+  override func unmount(
+    in reconciler: StackReconciler<R>,
+    with transaction: Transaction,
+    parentTask: UnmountTask<R>?
+  ) {
+    super.unmount(in: reconciler, with: transaction, parentTask: parentTask)
+
     guard let target = target else { return }
 
+    let task = UnmountHostTask(self, in: reconciler) {
+      self.mountedChildren.forEach {
+        $0.unmount(in: reconciler, with: transaction, parentTask: self.unmountTask)
+      }
+    }
+    task.isCancelled = parentTask?.isCancelled ?? false
+    unmountTask = task
+    parentTask?.childTasks.append(task)
     reconciler.renderer.unmount(
       target: target,
       from: parentTarget,
-      with: self
-    ) {
-      self.mountedChildren.forEach { $0.unmount(with: reconciler) }
-    }
+      with: task
+    )
   }
 
   override func update(in reconciler: StackReconciler<R>, with transaction: Transaction) {
     guard let target = target else { return }
 
+    // Stop any unfinished unmounts and complete them without transitions.
+    parentUnmountTask.cancel()
+    parentUnmountTask.completeImmediately()
+    parentUnmountTask = .init()
```

Comment:
If `scenePhase` changes to `background` then back to `active` (changing tabs, for example), you get a stack buffer overflow error. This happens even from the Demo main page. Commenting out these lines removes the error.

Reply:
This can be fixed with #367. I also removed the recursion from `UnmountTask`, maybe that will do something 🤷‍♂️